src/ProjectOrigin.WalletSystem.Server/Activities/TransferFullSliceActivity.cs

```diff
@@ -0,0 +1,152 @@
+
+public record TransferFullSliceArguments(Guid SourceSliceId, Guid ReceiverDepositEndpointId);
+
+public class TransferFullSliceActivity : IExecuteActivity<TransferFullSliceArguments>
+{
+    private readonly UnitOfWork _unitOfWork;
+    private readonly ILogger<TransferPartialSliceActivity> _logger;
+    private readonly IEndpointNameFormatter _formatter;
+
+    public TransferFullSliceActivity(
+        UnitOfWork unitOfWork,
+        ILogger<TransferPartialSliceActivity> logger,
+        IEndpointNameFormatter formatter)
+    {
+        _unitOfWork = unitOfWork;
+        _logger = logger;
+        _formatter = formatter;
+    }
+
+
+    public async Task<ExecutionResult> Execute(ExecuteContext<TransferFullSliceArguments> context)
+    {
+        _logger.LogTrace("RoutingSlip {TrackingNumber} - Executing {ActivityName}", context.TrackingNumber, context.ActivityName);
+
+        try
+        {
+            var sourceSlice = await _unitOfWork.CertificateRepository.GetSlice(context.Arguments.SourceSliceId);
+            var receiverDepositEndpoint = await _unitOfWork.WalletRepository.GetDepositEndpoint(context.Arguments.ReceiverDepositEndpointId);
+
+            var nextReceiverPosition = await _unitOfWork.WalletRepository.GetNextNumberForId(receiverDepositEndpoint.Id);
+            var receiverPublicKey = receiverDepositEndpoint.PublicKey.Derive(nextReceiverPosition).GetPublicKey();
+
+            var transferredSlice = new Slice(Guid.NewGuid(), receiverDepositEndpoint.Id, nextReceiverPosition, sourceSlice.RegistryId, sourceSlice.CertificateId, sourceSlice.Quantity, sourceSlice.RandomR, SliceState.Registering);
+            await _unitOfWork.CertificateRepository.InsertSlice(transferredSlice);
+
+            var registry = await _unitOfWork.RegistryRepository.GetRegistryFromId(sourceSlice.RegistryId);
+
+            var transferredEvent = CreateTransferEvent(registry.Name, sourceSlice, receiverPublicKey);
+
+            var sourceSlicePrivateKey = await _unitOfWork.WalletRepository.GetPrivateKeyForSlice(sourceSlice.Id);
+            var transaction = CreateAndSignTransaction(transferredEvent.CertificateId, transferredEvent, sourceSlicePrivateKey);
+
+            _unitOfWork.Commit();
+
+            var states = new Dictionary<Guid, SliceState>() {
+                { sourceSlice.Id, SliceState.Sliced },
+                { transferredSlice.Id, SliceState.Transferred }
+            };
+
+            return AddTransferRequiredActivities(context, receiverDepositEndpoint, transferredSlice, transaction, states);
+        }
+        catch (Exception ex)
+        {
+            _unitOfWork.Rollback();
+            _logger.LogError(ex, "Error sending transactions to registry");
+            return context.Faulted(ex);
+        }
+    }
+
+    private ExecutionResult AddTransferRequiredActivities(ExecuteContext context, DepositEndpoint receiverDepositEndpoint, Slice transferredSlice, Transaction transaction, Dictionary<Guid, SliceState> states)
```

Comment:
Are these executed in sequence after eachout or at the same time?

Reply:
They are executed in sequence, and the next is only executed if previous succeeds